### PR TITLE
Bugfix 4566

### PR DESCRIPTION
**Pull request by rsdavies**
### Description
The standard name of m01s00i023 has been corrected to surface_snow_amount. 

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: docs/src/whatsnew/dev.rst [hunk 1]
```diff
@@ -37,7 +37,7 @@ This document explains the changes made to Iris for this release
 🐛 Bugs Fixed
 =============
 
-#. N/A
+#. `@rsdavies`_ modified the CF complaint standard name for m01s00i023 :issue:`4566`
```
Comment (bjlittle) [inline docs/src/whatsnew/dev.rst:40]: @rsdavies Awesome, thanks for this.

BTW the docs being built in CI are throwing the error:
```shell
ERROR: Unknown target name: "@rsdavies".
```
To fix this, simply add a reference to your `@rsdavies` target at the bottom of the `dev.rst` document e.g., add the following:
```
.. comment
    Whatsnew author names (@github name) in alphabetical order. Note that,
    core dev names are automatically included by the common_links.inc:

.. _@rsdavies: https://github.com/rsdavies
```

Comment (bjlittle) [reply docs/src/whatsnew/dev.rst:40]: @rsdavies Also, have you signed the CLA?

If not, here's the Google Form https://scitools.org.uk/cla/v4/form

Complete that to sign the CLA and we'll do the rest :+1:

Comment (rsdavies) [reply docs/src/whatsnew/dev.rst:40]: Thanks Bill, I've signed the CLA.

Comment (rsdavies) [reply docs/src/whatsnew/dev.rst:40]: And I've fixed the error in dev.rst :)

**Comment by bjlittle** [comment]
We also just need to consult a third-party STASH expert on this change with to ensure it aligns with expectation

**Comment by rsdavies** [comment]
@bjlittle  I _think_ I've managed to rebase it. First time working with a forked repo!

**Comment by trexfeathers** [comment]
> We also just need to consult a third-party STASH expert on this change with to ensure it aligns with expectation

@bjlittle have you approached anyone yet. @rcomer suggests Martin Best at the UK Met Office might be a good shout.

**Comment by trexfeathers** [comment]
I have now reached out for a third-party opinion 🤞

**Review by trexfeathers — CHANGES_REQUESTED** [review]
Thanks very much @mo-martinbest for confirming that this change is correct.

@rsdavies: I'm very sorry for letting this go stale. What's New entries are now in [`latest.rst`](https://github.com/SciTools/iris/blob/c0153aee95d24180f8af01f8ce93bde48cb8f870/docs/src/whatsnew/latest.rst) (rather than `dev.rst`), so please could you make that change? Once that is done I will merge.

If you're busy with other things over the next week then we (the core development team) can make your suggested change ourselves, but we'd love to have you as the named contributer so hopefully you can fit this in 🙂

**Comment by rsdavies** [comment]
@trexfeathers I _think_ I've got the change to `latest.rst` done, had a little issue with the updating so it looks like I did it twice!

File: docs/src/whatsnew/latest.rst [hunk 2]
```diff
@@ -49,6 +49,7 @@ This document explains the changes made to Iris for this release
 ===========================
 
 #. `@rcomer`_ made :meth:`~iris.cube.Cube.aggregated_by` faster. (:pull:`4970`)
+#. `@rsdavies`_ modified the CF complaint standard name for m01s00i023 :issue:`4566`
```
Comment (rcomer) [inline docs/src/whatsnew/latest.rst:52]:
```suggestion
#. `@rsdavies`_ modified the CF compliant standard name for m01s00i023 :issue:`4566`
```

**Comment by trexfeathers** [comment]
I've just noticed that we also have this LBFC mapping:

https://github.com/SciTools/iris/blob/68a39a5f5f1edac43e16d35e97a29f350972ca0c/lib/iris/fileformats/um_cf_map.py#L1210

[LBFC `93`](https://reference.metoffice.gov.uk/um/fieldcode/_93) does indeed correspond to [`m01s00i023`](https://reference.metoffice.gov.uk/um/stash/_m01s00i023), so should the above mapping also be changed to reference `surface_snow_amount`?

**Comment by rsdavies** [comment]
I would say yes, if possible.

**Comment by trexfeathers** [comment]
> I would say yes, if possible.

OK, two sounds like consensus!

The change would literally be:

```diff
- CFName('snowfall_amount', None, 'kg m-2'): 93, 
+ CFName('surface_snow_amount', None, 'kg m-2'): 93,  
```

If you could do the honours?

**Comment by rsdavies** [comment]
Voila, [8f42c71](https://github.com/SciTools/iris/pull/4569/commits/8f42c71f644d50327288704ed1e78ae389d31bb5) sorted

File: lib/iris/fileformats/um_cf_map.py [hunk 3]
```diff
@@ -1207,7 +1207,7 @@
     CFName('sea_ice_thickness', None, 'm'): 687,
     CFName('sea_surface_elevation', None, 'm'): 608,
     CFName('snow_grain_size', None, '1e-6 m'): 1507,
-    CFName('snowfall_amount', None, 'kg m-2'): 93,
+    CFName('surface_snow_amount', None, '1e-6 m'): 93,
```
[the remark this answers is not in the document]
Comment (trexfeathers) [reply lib/iris/fileformats/um_cf_map.py:1210]: FWIW I believe [`surface_snow_amount`](https://cfconventions.org/Data/cf-standard-names/current/build/cf-standard-name-table.html#:~:text=surface_snow_amount) is measured in mass-per-area

[the remark this answers is not in the document]
Comment (rsdavies) [reply lib/iris/fileformats/um_cf_map.py]: Oh fail that was an awful copy paste error on my part! 
Fixed now

[the remark this answers is not in the document]
Comment (trexfeathers) [reply lib/iris/fileformats/um_cf_map.py]: You've backed out more than you meant to! The LBFC mapping is now completely unchanged.

**Comment by trexfeathers** [comment]
OK, we're in! Thanks for your patience.

**Pull request merged**